examples/lfric/eg15/matvec_opt.py

```diff
@@ -69,24 +69,32 @@
 
 '''
 from __future__ import print_function
+from psyclone.psyir.nodes import BinaryOperation
+from psyclone.psyir.transformations.matmul2code_trans import Matmul2CodeTrans
 from psyclone.psyir.backend.fortran import FortranWriter
 
-
 def trans(psy):
     '''PSyclone transformation script for the Dynamo0.3 API to optimise
     the matvec kernel for many-core CPUs. For the moment simply find
-    the first matvec kernel in the example and then print out its
-    PSyIR representation and output it as Fortran using the PSyIR
-    Fortran back-end.
+    the first matvec kernel in the example, transform the matmul
+    intrinsic to equivalant inline code and then print out its PSyIR
+    representation and output it as Fortran using the PSyIR Fortran
+    back-end.
 
     '''
+    matmul2code_trans = Matmul2CodeTrans()
     fortran_writer = FortranWriter()
+
     for invoke in psy.invokes.invoke_list:
         schedule = invoke.schedule
         for kernel in schedule.coded_kernels():
             if kernel.name.lower() == "matrix_vector_kernel_code":
                 kernel_schedule = kernel.get_kernel_schedule()
-                # optimisations will go here.
+                # Replace matmul with inline code
+                for bin_op in kernel_schedule.walk(BinaryOperation):
+                    if bin_op.operator is BinaryOperation.Operator.MATMUL:
+                        matmul2code_trans.apply(bin_op)
+                # Future optimisations will go here.
```

Comment:
pylint suggests using `sys.exit()` at line 105.

Reply:
Changed. It makes no difference in my version of pylint.